libraries/AP_InertialSensor/AP_InertialSensor_Backend.cpp

```diff
@@ -225,22 +225,28 @@ void AP_InertialSensor_Backend::_notify_new_gyro_raw_sample(uint8_t instance,
         _imu._last_raw_gyro[instance] = gyro;
 
         // apply the low pass filter
-        _imu._gyro_filtered[instance] = _imu._gyro_filter[instance].apply(gyro);
+        Vector3f gyro_filtered = _imu._gyro_filter[instance].apply(gyro);
+
+        // apply the notch filter
+        if (_gyro_notch_enabled()) {
+            gyro_filtered = _imu._gyro_notch_filter[instance].apply(gyro_filtered);
+        }
 
         // apply the harmonic notch filter
         if (gyro_harmonic_notch_enabled()) {
-            _imu._gyro_filtered[instance] = _imu._gyro_harmonic_notch_filter[instance].apply(_imu._gyro_filtered[instance]);
+            gyro_filtered = _imu._gyro_harmonic_notch_filter[instance].apply(gyro_filtered);
         }
 
-        // apply the notch filter
-        if (_gyro_notch_enabled()) {
-            _imu._gyro_filtered[instance] = _imu._gyro_notch_filter[instance].apply(_imu._gyro_filtered[instance]);
-        }
-        if (_imu._gyro_filtered[instance].is_nan() || _imu._gyro_filtered[instance].is_inf()) {
+        // if the filtering failed in any way then reset the filters and keep the old value
+        if (gyro_filtered.is_nan() || gyro_filtered.is_inf()) {
             _imu._gyro_filter[instance].reset();
             _imu._gyro_notch_filter[instance].reset();
             _imu._gyro_harmonic_notch_filter[instance].reset();
         }
+        else {
```

Comment:
tiny formatting thing but this line should be "} else {"

Reply:
Fixed, thanks